### PR TITLE
Weather, take 2

### PR DESCRIPTION
**Pull request by esqu1**

### CODE REVIEW

File: .app.id~ [hunk 1]
```diff
@@ -1 +0,0 @@
-79e0800fba37e7abc532724f409b72be
```
Comment (adelq) [inline .app.id~]: Use the `--amend` flag to actually merge with the previous commit. We'll rebase/squash this pull request so it doesn't show up.

File: README.md [hunk 2]
```diff
@@ -396,6 +396,33 @@ Returns all the available rooms on a given date in a specified building. Date is
...
+            <td>JSON</td>
+        </tr>
+        <tr>
+            <td>Parameters</td>
+            <td>None</td>
+        </tr>
+    </tbody>
+</table>
+
+#### <a href="weather.json">Example</a>
```
Comment (adelq) [inline README.md]: Nice

File: tests.py [hunk 3]
```diff
@@ -141,24 +141,31 @@ def test_extraction(self):
...
 
-  def test_invalid_token_validation(self):
+  def testInvalidTokenValidation(self):
     with server.app.test_request_context(headers=authHeaders):
       server.auth.auth()
       res = json.loads(server.auth.validate("badtoken").data.decode('utf8'))
       self.assertEquals(res['status'], 'invalid')
 
-  def test_token_validation_no_https(self):
+  def testTokenValidationNoHttps(self):
```
Comment (adelq) [inline tests.py]: Thanks for the consistency fix - not a fan, but consistency > style always

File: weather.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{"weather": {"coord":{"lon":-75.16,"lat":39.95},"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01d"}],"base":"stations","main":{"temp":62.49,"pressure":1022,"humidity":51,"temp_min":59,"temp_max":64.4},"visibility":16093,"wind":{"speed":9.01,"deg":300.001},"clouds":{"all":1},"dt":1478382660,"sys":{"type":1,"id":2361,"message":0.1837,"country":"US","sunrise":1478345747,"sunset":1478382735},"id":4560349,"name":"Philadelphia","cod":200}}
```
Comment (adelq) [inline weather.json]: Can this be formatted in a human readable way? There's a ton of online JSON formatters, alternatively use `jq`

File: tests.py [hunk 5]
```diff
@@ -141,24 +141,31 @@ def test_extraction(self):
       self.assertTrue("date" in d2[0])
       self.assertTrue("room_name" in d2[0])
 
+  def testWeather(self):
+    with server.app.test_request_context():
+      res = json.loads(server.weather.retrieve_weather_data().data.decode('utf8'))
+      self.assertTrue(len(res) > 0)
+      s = res['weather_data']
+      self.assertTrue("cod" in s)
```
Comment (adelq) [inline tests.py]: Where are the rest of the attribute tests? Tests are useful for finding out what data is expected in the output, definitely add back in.

File: server/weather.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import os
+
+
+@app.route('/weather/', methods=['GET'])
+def retrieve_weather_data():
+    '''Retrieves the current weather from the Open Weather Map API.
+    Stores data in a cache whenever data is retrieved; cache is updated
+    if it has not been updated within 10 minutes.
+    '''
+    APPID = os.getenv('APPID')
```
Comment (adelq) [inline server/weather.py]: Can this be something more specific? Something like "OWM_API_KEY". Naming is hard.

File: tests.py (hunk 5 above)
Comment (esqu1) [inline tests.py]: I took it out because the tests failed on Travis without the API key; the attributes do not load when the API key is nonexistent.

Comment (esqu1) [inline tests.py]: Is there a way to change the environment variables on Github?

Comment (adelq) [inline tests.py]: You have to set them on Travis, and I'm not sure if you can. Send me the key on Messenger and I'll take care of it.

File: tests.py [hunk 7]
```diff
@@ -141,24 +141,43 @@ def test_extraction(self):
       self.assertTrue("date" in d2[0])
       self.assertTrue("room_name" in d2[0])
 
+  def testWeather(self):
+    with server.app.test_request_context():
+      res = json.loads(server.weather.retrieve_weather_data().data.decode('utf8'))
+      self.assertTrue(len(res) > 0)
+      s = res['weather_data']
+      print s
```
Comment (adelq) [inline tests.py]: Remove print

Comment (esqu1) [inline tests.py]: oops

**Comment by adelq** [comment]
LGTM once tests pass

**Comment by esqu1** [comment]
That took a while

**Pull request merged**